Test schedules

blip2 opened the pull request:
opening a PR for discussion, would this be useful to provide a framework for testing?

blip2 committed:
first pass at test schedules for discussion

russss commented:
Oh yes, I wanted to do this! However there is a bit of complexity here, because we can't plausibly test every final circuit, and I don't want to auto-generate a schedule which implies that we should (there are easily hundreds of them).

Ideally we want the test schedule to automatically select distros which are most likely to be problematic, and generate a test schedule for those only.

blip2 commented:
Automatically or selected based on some metadata associated with specific nodes?
I would guess that high on the list might be:
- distros supplying equipment mounted on conductive structures which can be touched by members of the public (seating banks/staging/truss etc)
- distros serving zone 0/1 areas (showers)
- any distro serving non-hired equipment (and thus tested before coming to site?)
- other areas with higher (perceived or otherwise) risks (workshop, creche)
probably hard to establish these automatically but you could do a certain percentage for everything else?

russss commented:
Well, the aim is to make sure all our own infrastructure is safe, so we should be testing the distros furthest from the generator for Z<sub>s</sub> and earth continuity. This will give us confidence that the calculated fault loop impedances match up.

It may also be worth testing the end of each three-phase run for phase and neutral presence (and, less importantly given that we won't have any 3 phase motors on site, rotation). We've never had any problem with this with our suppliers, but I have seen the results of a lost neutral at another hacker event and it was a bit costly.

It should be possible to automatically identify those tests. Then I think it is probably worth manually identifying some important circuits (catering, stages, toilets/showers) and testing those. I don't know if it's worth automating this - it's probably just worth having space to write in some circuits manually.

blip2 committed:
calculate distance from source for each node

blip2 committed:
add rcd information to edge data

blip2 committed:
updated to provide revised test selection logic
- Test each adjustable RCD
- Test the end of each three phase run
- Test the distro fed from the longest submain on each grid
- Test control position supply from each powercube

blip2 commented:
Updated to provide this test selection logic
Template also updated so you can copy-paste into a G2 form

blip2 committed:
moved rcd value to type/rating column to allow for test values to be entered

blip2 committed:
tidy up and fixed longest distro selection

russss commented on dev-requirements.txt:
```diff
@@ -6,3 +6,4 @@ flake8==4.0.1
 black==22.3.0
 mypy==0.910
 types-PyYAML==6.0.4
+Pint==0.18
```
You shouldn't need this in dev-requirements - it's in setup.py. There's a version conflict here which is making the tests fail.

If you need to install the package for dev you can use `pip install -e .` which will also pull in the install requirements.

blip2 replied:
you fixed this properly in 90e49ecb - dropped this commit

blip2 committed:
Merge branch 'emfcamp:master' into test-schedules

russss committed:
Fix Black error

russss committed:
Fix flake8 errors

russss committed:
Fix mypy errors
